// @shipa-corp/pulumi/volumeBind.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
class VolumeBind extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["app"] = state ? state.app : undefined;
            inputs["mountPoint"] = state ? state.mountPoint : undefined;
            inputs["noRestart"] = state ? state.noRestart : undefined;
            inputs["volume"] = state ? state.volume : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.app === undefined) && !opts.urn) {
                throw new Error("Missing required property 'app'");
            }
            if ((!args || args.mountPoint === undefined) && !opts.urn) {
                throw new Error("Missing required property 'mountPoint'");
            }
            if ((!args || args.volume === undefined) && !opts.urn) {
                throw new Error("Missing required property 'volume'");
            }
            inputs["app"] = args ? args.app : undefined;
            inputs["mountPoint"] = args ? args.mountPoint : undefined;
            inputs["noRestart"] = args ? args.noRestart : undefined;
            inputs["volume"] = args ? args.volume : undefined;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(VolumeBind.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing VolumeBind resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VolumeBind(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VolumeBind.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VolumeBind.__pulumiType;
    }
}
exports.VolumeBind = VolumeBind;
/** @internal */
VolumeBind.__pulumiType = 'shipa:index/volumeBind:VolumeBind';
